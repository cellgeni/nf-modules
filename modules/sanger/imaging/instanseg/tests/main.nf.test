nextflow_process {

    name "Test Process IMAGING_INSTANSEG"
    script "../main.nf"
    process "IMAGING_INSTANSEG"

    tag "modules"
    tag "modules_sanger"
    tag "imaging"
    tag "imaging/instanseg"

    test("mouse_heart - tiff") {

        when {
            process {
                """
                input[0] = [
                    [ id:'mouse_heart'],
                    0, 0, 200, 200,
                    file(params.modules_testdata_base_path + 'imaging/tiff/mindagap.mouse_heart.wga.tiff', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("mouse_heart - tiff - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'mouse_heart'],
                    0, 0, 200, 200,
                    file(params.modules_testdata_base_path + 'imaging/tiff/mindagap.mouse_heart.wga.tiff', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}