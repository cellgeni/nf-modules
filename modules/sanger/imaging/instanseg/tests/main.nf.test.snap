{
    "mouse_heart - tiff - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart_0_0_200_200_instanseg_outlines.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,6587b1fbffd0ce3ddd72cb0dea70eeef"
                ],
                "versions": [
                    "versions.yml:md5,6587b1fbffd0ce3ddd72cb0dea70eeef"
                ],
                "wkts": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart_0_0_200_200_instanseg_outlines.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-02-11T15:25:06.256618571"
    },
    "mouse_heart - tiff": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart_0_0_200_200_instanseg_outlines.wkt:md5,59ac89147a5b5ffac7ca71dde15fbc9d"
                    ]
                ],
                "1": [
                    "versions.yml:md5,6587b1fbffd0ce3ddd72cb0dea70eeef"
                ],
                "versions": [
                    "versions.yml:md5,6587b1fbffd0ce3ddd72cb0dea70eeef"
                ],
                "wkts": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart_0_0_200_200_instanseg_outlines.wkt:md5,59ac89147a5b5ffac7ca71dde15fbc9d"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-02-11T15:24:37.887655481"
    }
}